src/python_testing/TC_DeviceBasicComposition.py

```diff
@@ -251,16 +251,47 @@ def fail_current_test(self, msg: Optional[str] = None):
             asserts.fail(msg)
 
     # ======= START OF ACTUAL TESTS =======
-    def test_endpoint_zero_present(self):
-        logging.info("Validating that the Root Node endpoint is present (EP0)")
+    def test_TC_SM_1_1(self):
+        self.print_step(1, "Perform a wildcard read of attributes on all endpoints - already done")
+        self.print_step(2, "Verify that endpoint 0 exists")
         if 0 not in self.endpoints:
             self.record_error(self.get_test_name(), location=AttributePathLocation(endpoint_id=0),
                               problem="Did not find Endpoint 0.", spec_location="Endpoint Composition")
             self.fail_current_test()
 
-    def test_descriptor_present_on_each_endpoint(self):
-        logging.info("Validating each endpoint has a descriptor cluster")
+        self.print_step(3, "Verify that endpoint 0 descriptor cluster includes the root node device type")
+        if Clusters.Descriptor not in self.endpoints[0]:
+            self.record_error(self.get_test_name(), location=AttributePathLocation(endpoint_id=0),
+                              problem="No descriptor cluster on Endpoint 0", spec_location="Root node device type")
+            self.fail_current_test()
+        root_dev_type = Clusters.Descriptor.Structs.DeviceTypeStruct(deviceType=0x16, revision=1)
```

Comment:
I would put a constant here for device type.

Reply:
do we want to hard-code a revision here? Should the test deny any deviceType 0x16 instead of just revision 1?

Reply:
I don't know, really. Maybe yeah, the revision should go away.